### PR TITLE
Add python examples to CI

### PR DESCRIPTION
**Pull request by noraabiakar**
* Add some log messages in python examples
* Add python examples to CI

### CODE REVIEW

File: scripts/travis/build.sh [hunk 1]
```diff
@@ -110,6 +110,10 @@ if [[ "${WITH_PYTHON}" == "true" ]]; then
     make pyarb -j4                                                           || error "building pyarb"
     progress "Python unit tests"
     python$PY $python_path/test/unit/runner.py -v2                           || error "running python unit tests (serial)"
+    python$PY $python_path/example/network_ring.py                           || error "running python network_ring example"
```
Comment (schmitts) [inline scripts/travis/build.sh]: maybe make use of https://github.com/arbor-sim/arbor/pull/1181 ?

**Review by bcumming — APPROVED** [review]
Looks good.

File: scripts/travis/build.sh (hunk 1 above)
Comment (noraabiakar) [reply scripts/travis/build.sh]: Is there an advantage to doing it that way?

Comment (schmitts) [reply scripts/travis/build.sh]: it would test that the examples can be used as "standalone" executables

**Pull request merged**